lib/net/imap/response_parser.rb

```diff
@@ -1101,48 +1101,28 @@ def header_list
       end
 
       # RFC3501 & RFC9051:
-      # header-fld-name = astring
+      #   header-fld-name = astring
+      #
+      # NOTE: Previously, Net::IMAP recreated the raw original source string.
+      # Now, it returns the decoded astring value.  Although this is technically
+      # incompatible, it should almost never make a difference: all standard
+      # header field names are valid atoms:
+      #
+      # https://www.iana.org/assignments/message-headers/message-headers.xhtml
       #
       # Although RFC3501 allows any astring, RFC5322-valid header names are one
       # or more of the printable US-ASCII characters, except SP and colon.  So
       # empty string isn't valid, and literals aren't needed and should not be
-      # used.  This syntax is unchanged by [I18N-HDRS] (RFC6532).
+      # used.  This is explicitly unchanged by [I18N-HDRS] (RFC6532).
       #
       # RFC5233:
-      # optional-field  =   field-name ":" unstructured CRLF
-      # field-name      =   1*ftext
-      # ftext           =   %d33-57 /          ; Printable US-ASCII
-      #                     %d59-126           ;  characters not including
-      #                                        ;  ":".
-      #
-      # Atom and quoted should be sufficient.
+      #     optional-field  =   field-name ":" unstructured CRLF
+      #     field-name      =   1*ftext
+      #     ftext           =   %d33-57 /          ; Printable US-ASCII
+      #                         %d59-126           ;  characters not including
+      #                                            ;  ":".
       #
-      # TODO: Use original source string, rather than decode and re-encode.
-      # TODO: or at least, DRY up this code with the send_command formatting.
-      def header_fld_name
-        case (str = astring)
-        when ""
-          warn '%s header-fld-name is an invalid RFC5322 field-name: ""' %
-            [self.class]
-          return '""'
-        when /[\x80-\xff\r\n]/n
-          warn "%s header-fld-name %p has invalid RFC5322 field-name char: %p" %
-            [self.class, str, $&]
-          # literal
-          return "{" + str.bytesize.to_s + "}" + CRLF + str
-        when /[^\x21-\x39\x3b-\xfe]/n
-          warn "%s header-fld-name %p has invalid RFC5322 field-name char: %p" %
-            [self.class, str, $&]
-          # invalid quoted string
-          return '"' + str.gsub(/["\\]/n, "\\\\\\&") + '"'
-        when /[(){ \x00-\x1f\x7f%*"\\]/n
-          # quoted string
-          return '"' + str.gsub(/["\\]/n, "\\\\\\&") + '"'
-        else
-          # atom
-          return str
-        end
-      end
+      alias header_fld_name astring
```

Comment:
FYI: Here's the "compatible" version:
```suggestion
      def header_fld_name
        assert_no_lookahead # not really necessary
        start = @pos
        astring             # parse the (decoded) astring
        @str[start..@pos]   # grab the raw (encoded) string 
      end
```

Reply:
This version is implemented in #217.